### PR TITLE
Provide public API for session creation

### PR DESCRIPTION
**Pull request by jreij**
This PR is an attempt at providing a public API allowing merchants to create and setup a session, which they can pass to drop-in/components.

### CODE REVIEW

File: sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    suspend fun createSession(
+        sessionModel: SessionModel,
+        configuration: Configuration,
+    ): Result<CheckoutSession> {
+        return CheckoutSessionInitializer(sessionModel, configuration).setupSession()
+    }
+
+    // TODO docs
+    // callback alternative for the suspend function
+    fun createSession(
```
Comment (OscarSpruit) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:28]: If we are giving a callback alternative it should also be without `CoroutineScope` IMO. Now it's basically the same thing, but we are calling 'launch'. WDYT?

File: sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import com.adyen.checkout.sessions.model.SessionModel
+import kotlinx.coroutines.CoroutineScope
+
+// TODO docs
+object CheckoutSessionProvider {
+    // TODO docs
+    suspend fun createSession(
+        sessionModel: SessionModel,
+        configuration: Configuration,
+    ): Result<CheckoutSession> {
```
Comment (OscarSpruit) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:22]: Not totally sure if we should expose `Result` publicly 🤔 Some people really dislike it

**Review by OscarSpruit — COMMENTED** [review]
I like the idea, think we should do it like this

File: sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt (hunk 2 above)
Comment (jreij) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:22]: Yeah i had a sealed class first but I removed it for some reason, will add it back 👍

File: sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt (hunk 1 above)
Comment (jreij) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:28]: How do you suggest we do it without using any `CoroutineScope`?

Comment (OscarSpruit) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:28]: With `ThreadManager` 😂 No, let's not do that. To be honest I'm not sure what the best way of doing this is

Comment (jreij) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/provider/CheckoutSessionProvider.kt:28]: 😂 ☠️

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**